### ISSUE TITLE
ESlint-plugin-compat doesn't seem to be able to accurately flag errors for methods on HTMLElements

### ISSUE THREAD

**Issue by gwyneplaine** [amilajack/eslint-plugin-compat#651]
It seems at the very least the plugin has trouble detecting callExpressions on HTMLElements.

```js
/* This errors as expected with:
HTMLElement.attachInternals() is not supported in iOS Safari 15.6-15.8eslint[compat/compat](https://github.com/amilajack/eslint-plugin-compat/blob/master/docs/rules/compat.md)
*/

globalThis.HTMLElement.attachInternals();

/* None of these cases error despite clearly either:
*  a) extending the `HTMLElement` object interface or 
*  b) straight up being an instance of HTMLElement 
*/
const x = globalThis.HTMLElement;
new x().attachInternals();

const foo = document.createElement('custom-element');
foo.attachInternals();

class y extends HTMLElement {
    constructor () {
        this.attachInternals();
    }
}

new y().attachInternals();
```
Looking through the source code, it seems like the algorithm we're using to match for failing rules in [lintMemberExpression](https://github.com/amilajack/eslint-plugin-compat/blob/506819bd19cbfee558abc5a6edbd57064a620037/src/helpers.ts#L135C17-L135C37) isn't able to match failingRules on these cases. 

```js
  const foo = document.createElement('custom-element');
  foo.attachInternals();
```
 The expected rule.object here is HTMLElement, rule.property `attachInternals()` 
 The evaluated node object name is `foo`, node property `attachInternals()` 
 This fails the failingRule check on [line 170](https://github.com/amilajack/eslint-plugin-compat/blob/506819bd19cbfee558abc5a6edbd57064a620037/src/helpers.ts#L170)

```js
class y extends HTMLElement {
    constructor () {
        this.attachInternals();
    }
}
```
The expected `protochainId` here is `HTMLElement.attachInternals()` the evaluated protochainId is `attachInternals()` 
It seems we're  unable to evaluate / not evaluating the superclass of the `ClassDeclaration` represented by the `ThisExpression`. This fails the failingRule check on [line 156](https://github.com/amilajack/eslint-plugin-compat/blob/506819bd19cbfee558abc5a6edbd57064a620037/src/helpers.ts#L156)

Notably I get the same issue with other callexpressions on HTMLElement such as [hidePopover](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/hidePopover) and [showPopover](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/showPopover)

Though given the above examples I've given, this may be a larger problem with matching failing rules on memberExpressions. 

I've created a minimal reproduction case here. 
This leverages the latest version of this package with `eslint@8.57.1`, notably the same issue occurs with `eslint@9.23.0`

**Comment by angelikatyborska** [comment]
I'm not sure if this is related, but I'm experiencing problems with usages of `Element.checkVisibility` not being reported.

I expected this code to trigger the error:

```js
const checkForOverflow = (target HTMLDivElement) => {
  const isVisible = target.checkVisibility();
};
```

but it doesn't.

This code also doesn't trigger it:

```js
(new HTMLElement()).checkVisibility();
/* or */
const div = document.createElement('div')
div.checkVisibility();
```

The only way to trigger that error is to write:
```
Element.checkVisibility();
/* or */
(new Element()).checkVisibility();
```

If this is not related to the original issue, let me know and I'll open a separate issue for this.
